src/main.c

```diff
@@ -1484,13 +1546,67 @@ static h2o_iovec_t on_extra_status(void *unused, h2o_globalconf_t *_conf, h2o_re
                                           " \"connections\": %d,\n"
                                           " \"max-connections\": %d,\n"
                                           " \"listeners\": %zu,\n"
-                                          " \"worker-threads\": %zu",
+                                          " \"worker-threads\": %zu,\n"
+                                          " \"num-sessions\": %" PRIu64,
                        SSLeay_version(SSLEAY_VERSION), current_time, restart_time, (uint64_t)(now - conf.launch_time), generation,
-                       num_connections(0), conf.max_connections, conf.num_listeners, conf.num_threads);
+                       num_connections(0), conf.max_connections, conf.num_listeners, conf.num_threads, num_sessions(0));
     assert(ret.len < BUFSIZE);
 
-    return ret;
+    if (JEMALLOC_STATS) {
```

Comment:
Could you please change this to `#if JEMALLOC_STATS`? The reason I suggested this is since there is not guarantee that a compiler strips non-reachable code (i.e. the block inside `if (0) { }`). In such case, reference to `malloc_stats_print` would cause a link-time error.